compiler/src/dotty/tools/dotc/transform/sjs/JSInteropUtils.scala

```diff
@@ -0,0 +1,78 @@
+package dotty.tools.dotc
+package transform
+package sjs
+
+import core._
+import util.SrcPos
+import Annotations._
+import Constants._
+import Contexts._
+import Decorators._
+import DenotTransformers._
+import Flags._
+import NameOps._
+import Names._
+import Phases._
+import Scopes._
+import StdNames._
+import Symbols._
+import SymDenotations._
+import SymUtils._
+import ast.Trees._
+import Types._
+
+import dotty.tools.backend.sjs.JSDefinitions.jsdefn
+
+object JSInteropUtils {
+  enum JSName {
+    case Literal(name: String)
+    case Computed(sym: Symbol)
+
+    def displayName(using Context): String = this match {
+      case Literal(name) => name
+      case Computed(sym) => sym.fullName.toString()
+    }
+  }
+
+  extension (sym: Symbol) {
+    /** Should this symbol be translated into a JS getter? */
+    def isJSGetter(using Context): Boolean = {
+      sym.is(Module) || (
+          sym.is(Method)
+            && sym.info.firstParamTypes.isEmpty
+            && atPhaseNoLater(erasurePhase)(sym.info.isParameterless))
```

Comment:
Why is the logic different form `dotty.tools.backend.sjs.JSInterop.isJSGetter`.

Reply:
It's different because the logic that is (now was) in `JSInterop` is only correct after erasure. The new logic in `JSInteropUtils` is valid both before and after erasure.